node/actors/executor/src/attestation.rs

```diff
@@ -0,0 +1,89 @@
+//! Module to publish attestations over batches.
+
+use std::sync::Arc;
+
+use anyhow::Context;
+use zksync_concurrency::ctx;
+use zksync_concurrency::time;
+use zksync_consensus_network::gossip::BatchVotesPublisher;
+use zksync_consensus_roles::attester::BatchNumber;
+use zksync_consensus_storage::{BatchStore, BlockStore};
+
+use crate::Attester;
+
+const POLL_INTERVAL: time::Duration = time::Duration::seconds(1);
+
+/// Polls the database for new batches to be signed and publishes them to the gossip channel.
+pub(super) struct AttesterRunner {
+    block_store: Arc<BlockStore>,
+    batch_store: Arc<BatchStore>,
+    attester: Attester,
+    publisher: BatchVotesPublisher,
+}
+
+impl AttesterRunner {
+    /// Crete a new instance of a runner.
+    pub(super) fn new(
+        block_store: Arc<BlockStore>,
+        batch_store: Arc<BatchStore>,
+        attester: Attester,
+        publisher: BatchVotesPublisher,
+    ) -> Self {
+        Self {
+            block_store,
+            batch_store,
+            attester,
+            publisher,
+        }
+    }
+    /// Poll the database for new L1 batches and publish our signature over the batch.
+    pub(super) async fn run(self, ctx: &ctx::Ctx) -> ctx::Result<()> {
+        let public_key = self.attester.key.public();
+        // The first batch number we want to publish our vote for. We don't have to re-publish a vote
+        // because once it enters the vote register even future peers can pull it from there.
+        let mut min_batch_number = BatchNumber(0);
+        loop {
+            // Pretend that the attesters can evolve.
+            let Some(attesters) = self.block_store.genesis().attesters.as_ref() else {
```

Comment:
not very relevant now, but doesn't this one need to pass the l1 batch number for which we need the list of attesters?

Reply:
Today was the first time I heard about this plan to have attesters be queries that way. I knew looking for attesters in genesis probably wasn't going to be the final form, but it seems pervasive in the codebase, so I didn't see a reason to start deviating from it here. The same goes for the list of validators. 

What I thought was that there might be some kind of configuration that lives on until it's replaced with a new one, those _epochs_ that we saw in the Solidity code, so that one might have to know the current and previous and potentially the next epochs, but not arbitrary batch/block numbers.

This is potentially one of those places where an STM approach would help as well, to have an in-memory representation of what we believe is _the_ current state, without having to deal with eventual consistency between every component of the system.

So you are right that this is a design smell, but it's such a gaping hole that I don't know what to fill it with at the moment that looks more plausible.

Reply:
Makes sense... So maybe we should request it one and not in a loop since it will not work correctly in the loop in its current form..

Reply:
For now I think it's fine to do it like this. We are going to deploy first without attester rotation anyway. There's a lot of places where we use genesis() so it's better if we do it separately.

Reply:
This was moved out of the loop in #150